### ISSUE TITLE
Allow nil for enums

### ISSUE THREAD

**Issue by andyweiss1982** [neo4jrb/neo4j#1261]
It seems the current enum implementation default the model's property to the first declared value if none is specified.  For example:

```
class Service
  include Neo4j::ActiveNode

  enum service_type: {
    foo: 0,
    bar: 1,
  }
```

Will mean that `Service.create` will return a Foo Service by default, even if no `service_type` is given.  

Ideally, I would like to allow a user to select from a list and fail a validation if no selection is given.  

Is there a way I can get the enum to default to `nil` instead of `0` in this case?

**Comment by ProGM** [comment]
@andyweiss1982 Hi, I'm trying to fix this!
What version of neo4j gem are you using?

If you're using 7.0.x, try this branch please ;)

```ruby
gem 'neo4j', github: 'neo4jrb/neo4j', branch: 'fix-enum-default-value'
```

Thank you very much!

**Comment by andyweiss1982** [comment]
Hi @ProGM Thanks!!

This is what we were using: 

```
GIT
  remote: git://github.com/neo4jrb/neo4j.git
  revision: 6ff240c04381f18f2a3dd19a014ba2149aef6f65
  branch: rails5
  specs:
    neo4j (7.0.3)
      activemodel (>= 4)
      activesupport (>= 4)
      neo4j-core (>= 6.0.0)
      orm_adapter (~> 0.5.0)
```

Do you think we are in the clear to make the switch to the new branch?

**Comment by andyweiss1982** [comment]
@ProGM It looks like I have a conflict with rails 5 / active support.  However I am happy to spin up a rails 4.2 demo app to test out the branch if you like.

```
Bundler could not find compatible versions for gem "activesupport":
  In Gemfile:
    rails (~> 5.0) was resolved to 5.0.0, which depends on
      activejob (= 5.0.0) was resolved to 5.0.0, which depends on
        globalid (>= 0.3.6) was resolved to 0.3.7, which depends on
          activesupport (>= 4.1.0)

    neo4j was resolved to 7.0.14, which depends on
      activesupport (~> 4)

    rails (~> 5.0) was resolved to 5.0.0, which depends on
      activesupport (= 5.0.0)

    rails (~> 5.0) was resolved to 5.0.0, which depends on
      activesupport (= 5.0.0)

    rails (~> 5.0) was resolved to 5.0.0, which depends on
      activesupport (= 5.0.0)

    rails (~> 5.0) was resolved to 5.0.0, which depends on
      activesupport (= 5.0.0)

    rails (~> 5.0) was resolved to 5.0.0, which depends on
      activesupport (= 5.0.0)

    rails (~> 5.0) was resolved to 5.0.0, which depends on
      activesupport (= 5.0.0)

    rspec-rails (~> 3.0) was resolved to 3.5.1, which depends on
      activesupport (>= 3.0)
```

**Comment by ProGM** [comment]
Oh, you're right! 7.0.x branch was for rails4 only.

I created a new branch for neo4j 7.1.x, that supports rails5 ;)

Try adding this:
```
gem 'neo4j', github: 'neo4jrb/neo4j', branch: 'fix-enum-default-value-7.1.x'
```

**Comment by cheerfulstoic** [comment]
That branch should be good for testing, though when we merge we can merge the one off 7.0.x and then I'll merge it forward from `7.0.x` to `7.1.x` and release another patch version

**Comment by andyweiss1982** [comment]
@ProGM I was able to get my specs to pass without complication.  I am now able to implement a much more straighforward presence validation on my enum properties.  Thank you very much - let me know if there is any more info you need from me!

```
GIT
  remote: git://github.com/neo4jrb/neo4j.git
  revision: ae3af43f6079a38d8f3c0a40258bf43b4d6933ff
  branch: fix-enum-default-value-7.1.x
  specs:
    neo4j (7.1.2)
      activemodel (>= 4.0, < 5.1)
      activesupport (>= 4.0, < 5.1)
      neo4j-core (< 7.0.0)
      orm_adapter (~> 0.5.0)
```

```
123 examples, 0 failures
```

**Comment by ProGM** [comment]
Great 👍 
Thank you very much!

@cheerfulstoic Now merging, I think we can release it ;P

**Comment by cheerfulstoic** [comment]
Released as 7.0.15 and 7.1.3.  Thanks all!

**Comment by ProGM** [comment]
@cheerfulstoic  it should be merged on master and 8.x as well, I think ;P
